plugins/urls/__init__.py

```diff
@@ -178,113 +219,139 @@ def message_handler(self, event=MessageReceived):
+        matches = extract_urls(message)
 
-                if url in ["http://", "https://"]:
-                    self.logger.trace(_("URL is not actually a URL, just %s"
-                                        % url))
-                    return
+        for match in matches:
+            self.logger.trace("match: {0}", match)
 
-                if self.check_blacklist(url):
-                    self.logger.debug(_("Not parsing, URL is blacklisted."))
-                    return
+            _url = self.match_to_url(match)
 
-                if isinstance(target, Channel):
-                    try:
-                        self.catcher.insert_url(url, source.nickname,
-                                                target.name, protocol.name)
-                    except Exception:
-                        self.logger.exception(_("Error catching URL"))
-
-                title, domain = self.parse_title(url)
-
-                self.logger.trace(_("Title: %s") % title)
-
-                if isinstance(target, Channel):
-                    if protocol.name not in self.channels:
-                        with self.channels:
-                            self.channels[protocol.name] = {
-                                target.name: {"last": url,
-                                              "status": "on",
-                                              "shortener":
-                                              "tinyurl"}
-                            }
-                    if target.name not in self.channels[protocol.name]:
-                        with self.channels:
-                            self.channels[protocol.name][target.name] = {
-                                "last": url,
-                                "status": "on",
-                                "shortener":
-                                "tinyurl"
-                            }
-                    else:
-                        with self.channels:
-                            self.channels[protocol.name][target.name]["last"] \
-                                = url
-                    if title is None:
-                        return
+            if _url is None:
+                continue
 
-                    if domain is not None and "/" in domain:
-                        domain = domain.split("/")[0]
-                    if domain is None:
-                        target.respond(title)
-                    else:
-                        target.respond("\"%s\" at %s" % (title, domain))
-                elif isinstance(target, User):
-                    if title is None:
-                        return
+            # Check redirects, following as necessary
+
+            redirects = 0
+            max_redirects = self.config.get("redirects", {}).get("max", 15)
+            domains = self.config.get("redirects", {}).get("domains", [])
+
+            self.logger.debug("Checking redirects...")
+
+            while _url.domain in domains and redirects < max_redirects:
```

Comment:
...yes, and my question was, what happens when that hits max_redirects (i.e. when that fucking whiel loop exits), and why isn't your answer "it stops processing the URL"?